### PR TITLE
Use new uint version

### PR DESCRIPTION
**Pull request by dvdplm**
BREAKING CHANGE: Remove U1024 and H1024 
Remove conversion facilities that moved to uint 
Remove asm 
Cleanup

Depends on https://github.com/paritytech/parity-common/pull/30

### CODE REVIEW

File: ethereum-types/Cargo.toml [hunk 1]
```diff
@@ -1,22 +1,18 @@
...
 
 [dependencies]
-crunchy = "0.1.5"
+crunchy = "0.1"
 ethbloom = { path = "../ethbloom", version = "0.5.0", default-features = false }
 ethereum-types-serialize = { version = "0.2.1", path = "../serialize", optional = true }
 fixed-hash = { git = "https://github.com/paritytech/parity-common", default_features = false }
 serde = { version = "1.0", optional = true }
-uint = { git = "https://github.com/paritytech/parity-common", default_features = false }
+uint = { version = "0.4.1", git = "https://github.com/paritytech/parity-common", default_features = false }
```
Comment (NikVolf) [inline ethereum-types/Cargo.toml]: why not from crates.io?

Comment (debris) [reply ethereum-types/Cargo.toml]: we should use either `version =` or `git =`. Never both of them

Comment (dvdplm) [reply ethereum-types/Cargo.toml]: It is not published yet. My plan was to let the dust settle and then go through each crate in `parity-common` and publish them properly. At the moment it's a bit inconsistent and fluctuating.

Comment (NikVolf) [reply ethereum-types/Cargo.toml]: use git dependency until then?

Comment (NikVolf) [reply ethereum-types/Cargo.toml]: as @debris pointed out, without version

Comment (dvdplm) [reply ethereum-types/Cargo.toml]: done

**Review by debris — APPROVED** [review]

**Pull request merged**